simple-auth/src/main/java/com/fga/example/fga/PostOpenFgaCheck.java

```diff
@@ -0,0 +1,24 @@
+package com.fga.example.fga;
+
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.access.prepost.PostAuthorize;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@PostAuthorize("@openFga.check({object}, {objectType}, {relation}, {userType}, {userId})")
+public @interface PostOpenFgaCheck {
+    String object();
+    String relation();
+    String userType();
+    String objectType();
+
+    String userId() default "authentication.name";
```

Comment:
It is available in 6.2.

So if we look at the final result, we're just mapping the following
```java
@PreAuthorize("@openFga.check({object}, {objectType}, {relation}, {userType}, {userId})"
```
`PreAuthorize` has got the full power of [SpEL](https://docs.spring.io/spring-framework/reference/core/expressions.html), and Spring Security just injects `authentication` as a keyword.

I was on the fence between using this and some sort of function, since there is a minor overhead cost to SpEL, but SpEL works great.